addons/sale_mrp/controllers/portal.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request
+from odoo.addons.portal.controllers.portal import CustomerPortal
+
+
+class SaleMrpPortal(CustomerPortal):
+    def _get_additional_order_page_values(self, sale_order=None):
+        """ Override of sale to include the MOs linked to the sale.order into its portal page. """
+        rendering_context_values = super()._get_additional_order_page_values(sale_order)
+        if sale_order:
+            procurement_groups = request.env['procurement.group'].sudo().search([('sale_id', '=', sale_order.id)])
+            mrp_production_ids = procurement_groups.stock_move_ids.created_production_id.procurement_group_id.mrp_production_ids | procurement_groups.mrp_production_ids
```

Comment:
It's probably time to clean it and add mrp_production_ids field on SO so we don't duplicate this code everywhere

Reply:
Done. Updated the existing `_compute_mrp_production_count()` to handle the calculation of both `mrp_production_ids` and `mrp_production_count`.